### PR TITLE
fix: [#185337472] Trade Name businesses should not have a Business Effective Date

### PR DESCRIPTION
**Pull request by cbrunefearless**
## Description

Trade Name businesses should not have a Business Effective Date label and should not have a Business Effective Date picker under Business Profile.
### Ticket

[185337472](https://www.pivotaltracker.com/story/show/185337472)

### Approach

No changes to dependencies
### Steps to Test
Unit tests to check for:

1. does display date of formation question when legal structure is undefined
2. does not display date of formation question when legal structure is defined and has a trade name
3. does display date of formation question when legal structure is defined and does not have trade name

**Test as existing business with no legal structure defined**
1. Log in as an existing business with no legal structure defined.
2. Click on the business profile under your name in the top right hand corner
3. Label on top should show "Business Effective Date".
4. Click on the Label "Business Effective Date" and it should take you to the "Business Effective Date" date picker.

**Test as existing business with a legal structure defined that has a trade name**
1. Log in as an existing business with a legal structure defined and has a trade name associated with it (sole-proprietorship or general-partnership).
2. Click on the business profile under your name in the top right hand corner
3. Label on top should not show "Business Effective Date" and should not show "Business Effective Date" date picker.

**Test as existing business with a legal structure defined that has no trade name**
1. Log in as an existing business with a legal structure defined and has no trade name associated with it (limited-partnership).
2. Click on the business profile under your name in the top right hand corner
3. Label on top should show "Business Effective Date".
4. Click on the Label "Business Effective Date" and it should take you to the "Business Effective Date" date picker.

## Code author checklist

- [X] I have performed a self-review of my code
- [X] I have created and/or updated relevant documentation, if necessary
- [X] I have not used any relative imports
- [X] I have checked for and removed instances of unused config from CMS
- [X] I have pruned any instances of unused code
- [X] I have not added any markdown to labels, titles and button text in config

### CODE REVIEW

File: web/src/components/profile/ProfileDateOfFormation.tsx [hunk 1]
```diff
@@ -70,7 +70,7 @@ export const ProfileDateOfFormation = (props: Props): ReactElement => {
     process.env.NODE_ENV === "test" || process.env.CI === "true" ? DesktopDatePicker : DatePicker;
 
   return (
-    <LocalizationProvider dateAdapter={AdapterDayjs}>
+    <LocalizationProvider dateAdapter={AdapterDayjs} data-testid="profile-date-of-formation-question">
```
Comment (aloverso) [inline web/src/components/profile/ProfileDateOfFormation.tsx:73]: Am I correct that this testId is never used? If so, it should be removed

File: web/test/pages/profile.test.tsx [hunk 2]
```diff
@@ -2319,10 +2320,53 @@ describe("profile", () => {
...
+          operatingPhase: "UP_AND_RUNNING_OWNING",
+          businessPersona: "OWNING",
+        },
+      });
+      renderPage({ userData });
+      expect(screen.getByTestId("question-dateOfFormation-alert-text")).toBeInTheDocument();
+      expect(screen.getByTestId("effective-date")).toBeInTheDocument();
+    });
+
+    it("does not display date of formation question when legal structure is defined and has a trade name", () => {
```
Comment (aloverso) [inline web/test/pages/profile.test.tsx:2336]: For this and the test below, I would rephrase to clarify, we think of legal structures as "being" a Trade Name type or a Public Filing type.

Saying "has" trade name is a little misleading, because `tradeName` is actually a completely separate field in the profile that businesses can enter their trade name.

I think what you mean to say here is "when it is a Trade Name legal structure" or similar

File: web/test/pages/profile.test.tsx [hunk 3]
```diff
@@ -2319,10 +2320,53 @@ describe("profile", () => {
...
+      const userData = generateUserData({
+        profileData: {
+          ...emptyProfileData,
+          operatingPhase: "UP_AND_RUNNING_OWNING",
+          businessPersona: "OWNING",
+          legalStructureId: "general-partnership",
```
Comment (aloverso) [inline web/test/pages/profile.test.tsx:2342]: we should use our randomized helpers here: `randomTradeNameBusiness` instead of only testing one case

File: web/test/pages/profile.test.tsx [hunk 4]
```diff
@@ -2319,10 +2320,53 @@ describe("profile", () => {
...
+      expect(screen.queryByTestId("effective-date")).not.toBeInTheDocument();
+    });
+
+    it("does display date of formation question when legal structure is defined and does not have trade name", () => {
+      const userData = generateUserData({
+        profileData: {
+          ...emptyProfileData,
+          operatingPhase: "UP_AND_RUNNING_OWNING",
+          businessPersona: "OWNING",
+          legalStructureId: "limited-partnership",
```
Comment (aloverso) [inline web/test/pages/profile.test.tsx:2356]: we should use our randomized helpers here: `randomPublicFilingLegalStructure` instead of only testing one case

File: web/test/pages/profile.test.tsx [hunk 5]
```diff
@@ -2319,10 +2320,53 @@ describe("profile", () => {
...
+      const userData = generateUserData({
+        profileData: {
+          ...emptyProfileData,
+          operatingPhase: "UP_AND_RUNNING_OWNING",
+          businessPersona: "OWNING",
+        },
+      });
+      renderPage({ userData });
+      expect(screen.getByTestId("question-dateOfFormation-alert-text")).toBeInTheDocument();
+      expect(screen.getByTestId("effective-date")).toBeInTheDocument();
```
[the remark this answers is not in the document]
Comment (jphechter) [reply web/test/pages/profile.test.tsx:2333]: The text for both of these is "Business Effective Date" which is why we opted for the `testId` route.

I don't see a aria-label on this right now. If we do want to use `getByLabelText` that might require modifying the component a little more.

[the remark this answers is not in the document]
Comment (jphechter) [reply web/test/pages/profile.test.tsx:2333]: Wasn't really looking that closely before, but I think there is a copy flub here:

Screenshot 2023-06-16 at 11 55 43 AM

File: web/src/components/profile/ProfileDateOfFormation.tsx (hunk 1 above)
Comment (cbrunefearless) [reply web/src/components/profile/ProfileDateOfFormation.tsx:73]: Yes, removed

File: web/test/pages/profile.test.tsx (hunk 2 above)
Comment (cbrunefearless) [reply web/test/pages/profile.test.tsx:2336]: changed the naming around that.  It was confusing for me at first to see it on the legal structure and the profle data.

File: web/test/pages/profile.test.tsx (hunk 3 above)
Comment (cbrunefearless) [reply web/test/pages/profile.test.tsx:2342]: using randomTradeNameLegalStructure

File: web/test/pages/profile.test.tsx (hunk 4 above)
Comment (cbrunefearless) [reply web/test/pages/profile.test.tsx:2356]: using randomPublicFilingLegalStructure

File: web/test/pages/profile.test.tsx [hunk 6]
```diff
@@ -2348,7 +2392,9 @@ describe("profile", () => {
     it("removes question from alert when it gets answered", () => {
       const userData = generateUserData({
         profileData: generateProfileData({
+          ...emptyProfileData,
           operatingPhase: "UP_AND_RUNNING_OWNING",
+          businessPersona: "OWNING",
```
[the remark this answers is not in the document]
Comment (cbrunefearless) [reply web/test/pages/profile.test.tsx:2397]: took out those changes

**Comment by aloverso** [comment]
@cbrunefearless also as a heads up before finishing, I see you have a merge commit on here - we typically use rebases to ensure that our main branch is a single source of truth. You'll want to make sure you only have 1 new commit (no merge commit), and then can be rebased (not merged) into main

File: web/test/pages/profile.test.tsx [hunk 7]
```diff
@@ -2327,6 +2387,7 @@ describe("profile", () => {
           existingEmployees: undefined,
           municipality: undefined,
           homeBasedBusiness: undefined,
+          legalStructureId: randomPublicFilingLegalStructure(),
```
Comment (cbrunefearless) [inline web/test/pages/profile.test.tsx:2390]: Had to change it to have only randomPublicFilingLegalStructure() because some of the legal structures would make it fail.

File: web/test/pages/profile.test.tsx [hunk 8]
```diff
@@ -2351,6 +2412,7 @@ describe("profile", () => {
           operatingPhase: "UP_AND_RUNNING_OWNING",
           dateOfFormation: undefined,
           existingEmployees: undefined,
+          legalStructureId: randomPublicFilingLegalStructure(),
```
Comment (cbrunefearless) [inline web/test/pages/profile.test.tsx:2415]: Had to change it to have only randomPublicFilingLegalStructure() because some of the legal structures would make it fail.
